http/src/util/service/context.rs

```diff
@@ -198,19 +198,16 @@ pub mod object {
 
     pub struct ContextObjectConstructor<Req, C>(PhantomData<(Req, C)>);
 
-    pub type ContextFactoryObject<Req: 'static, C: 'static, BErr, Res, Err> =
+    pub type ContextFactoryObject<Req, C, BErr, Res, Err> =
         impl BuildService<Error = BErr, Service = ContextServiceObject<Req, C, Res, Err>>;
 
-    pub type ContextServiceObject<Req: 'static, C: 'static, Res, Err> =
-        impl for<'c, 's> Service<&'c mut Context<'s, Req, C>, Response = Res, Error = Err>;
+    pub type ContextServiceObject<Req, C, Res, Err> =
+        impl for<'c> Service<Context<'c, Req, C>, Response = Res, Error = Err>;
 
     impl<C, I, Svc, BErr, Req, Res, Err> ObjectConstructor<I> for ContextObjectConstructor<Req, C>
     where
-        I: BuildService<Service = Svc, Error = BErr>,
-        Svc: for<'c, 's> Service<&'c mut Context<'s, Req, C>, Response = Res, Error = Err> + 'static,
-        I: 'static,
-        C: 'static,
-        Req: 'static,
+        I: BuildService<Service = Svc, Error = BErr> + 'static,
+        Svc: for<'c> Service<Context<'c, Req, C>, Response = Res, Error = Err> + 'static,
```

Comment:
`Req` is just assumed to be `'static`.

Can you give the test where it failed?

Reply:
```rust
    #[tokio::test]
    async fn lifetime() {
        struct Req<'a> {
            _r: &'a str
        }

        impl BorrowReq<http::Uri> for Req<'_> {
            fn borrow(&self) -> &http::Uri {
                todo!()
            }
        }

        async fn handler(req: &mut Context<'_, Req<'_>, String>) -> Result<Response<()>, Infallible> {
            let (_, state) = req.borrow_parts_mut();
            assert_eq!(state, "string_state");
            Ok(Response::new(()))
        }

        let router = GenericRouter::with_custom_object::<super::object::ContextObjectConstructor<_, _>>()
            .insert("/", fn_service(handler));

        let service = ContextBuilder::new(|| async { Ok::<_, Infallible>(String::from("string_state")) })
            .service(router)
            .build(())
            .await
            .ok()
            .unwrap();

        let r = String::new();

        let res = service.call(Req { _r: r.as_str() } ).await.unwrap();

        assert_eq!(res.status().as_u16(), 200);
    }
```

Removing it does not fix this test.